### PR TITLE
Cav 61

### PR DESCRIPTION
**Pull request by mar36tin**

### CODE REVIEW

**Comment by johno1985** [comment]
please can you squash the commits into a single one?

File: README.md [hunk 1]
```diff
@@ -1,8 +1,34 @@
...
+### Testing
+#### Unit tests
+`sbt clean test`
+
+#### Integration tests
+`sbt clean it:test`
+
+### Running app
+
+In order to run this microservice cip-phone-number you willl need to run the downstream services first. Then run
```
Comment (johno1985) [inline README.md:21]: is it worth mention service manager here?

File: build.sbt [hunk 2]
```diff
@@ -18,10 +18,9 @@ lazy val microservice = Project(appName, file("."))
...
       "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
     )
-    // ***************
   )
   .settings(publishingSettings: _*)
   .configs(IntegrationTest)
   .settings(integrationTestSettings(): _*)
   .settings(resolvers += Resolver.jcenterRepo)
-  .settings(CodeCoverageSettings.settings: _*)
+  .settings(CodeCoverageSettings.settings: _*)
```
Comment (johno1985) [inline build.sbt:26]: no new line at end of file

File: conf/application.conf [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 include "backend.conf"
 
 # This is the main configuration file for the application.
 # ~~~~~
 
 appName = cip-phone-number
+appUrl = "https://"${appName}".protected.mdtp"
```
Comment (johno1985) [inline conf/application.conf:21]: is this needed?

File: test/resources/correct_phonenumber.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "phoneNumber" : "07843274323"
+}
```
Comment (johno1985) [inline test/resources/correct_phonenumber.json:3]: no new line at end of file

File: test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+package uk.gov.hmrc.cipphonenumber.models
+
+import play.api.libs.json.{Json, Reads, __}
+
+case class PhoneNumber(phoneNumber: String)
+
+object PhoneNumber {
+  implicit val format = Json.format[PhoneNumber]
+
+  val phoneNumberReads: Reads[PhoneNumber] =
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala:26]: can this be moved into the unit test?

File: test/uk/gov/hmrc/cipphonenumber/utility/FileLoader.scala [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cipphonenumber.utility
+
+import play.api.libs.json.Json
+
+import scala.io.Source
+
+trait FileLoader {
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumber/utility/FileLoader.scala:23]: is this being used?

**Review by johno1985 — COMMENTED** [review]
all files should have one empty line at end of file

**Comment by johno1985** [comment]
there are also conflicts, I believe you need to rebase with latest changes from origin

File: README.md (hunk 1 above)
Comment (mar36tin) [reply README.md:21]: Yes we could do, can always add that later

File: conf/application.conf (hunk 3 above)
Comment (mar36tin) [reply conf/application.conf:21]: yes

Comment (johno1985) [reply conf/application.conf:21]: I can't see this property in the other repos. I also cannot see any usages of it in the code

File: app/uk/gov/hmrc/cipphonenumber/connectors/ValidateConnector.scala [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      .transform(_.withRequestFilter(AhcCurlRequestLogger()))
+      .execute[HttpResponse]
+
+    res flatMap parseResponse recoverWith {
+      case e: Throwable =>
+        logger.error(s"Downstream call failed: ${config.validateUrlHost}")
+        Future.failed(e)
+    }
+  }
+}
```
Comment (johno1985) [inline app/uk/gov/hmrc/cipphonenumber/connectors/ValidateConnector.scala:55]: no new line

File: build.sbt (hunk 2 above)
Comment (johno1985) [reply build.sbt]: still no new line

File: test/requests.rest [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+### 7-digit number
+POST http://localhost:8080/customer-insight-platform/phone-number/validate-format
+content-type: application/json
+
+{
+  "phoneNumber" : "0123456"
+}
+
+
```
Comment (johno1985) [inline test/requests.rest:41]: too many empty lines, only 1 needed

File: test/uk/gov/hmrc/cipphonenumber/connectors/ValidateConnectorSpec.scala [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        "microservice.services.cipphonenumber.validation.protocol" -> wireMockProtocol,
+      ))
+    )
+
+    val validateConnector = new ValidateConnector(
+      httpClientV2,
+      appConfig
+    )
+  }
+}
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumber/connectors/ValidateConnectorSpec.scala:79]: no new line

File: test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+package uk.gov.hmrc.cipphonenumber.models
+
+import play.api.libs.json.{Json, Reads, __}
+
+case class PhoneNumber(phoneNumber: String)
+
+object PhoneNumber {
+  implicit val format = Json.format[PhoneNumber]
+}
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala]: no new line

**Review by johno1985 — COMMENTED** [review]
comments added

File: test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala (hunk 5 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala]: No I think its best to just leave it there

**Review by brdgeth — DISMISSED** [review]

File: test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala (hunk 5 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumber/models/PhoneNumber.scala]: done

File: test/uk/gov/hmrc/cipphonenumber/utility/FileLoader.scala (hunk 6 above)
Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumber/utility/FileLoader.scala]: removed

File: README.md [hunk 11]
```diff
@@ -1,8 +1,35 @@
...
+### Testing
+#### Unit tests
+`sbt clean test`
+
+#### Integration tests
+`sbt clean it:test`
+
+### Running app
+
+In order to run this microservice cip-email you willl need to run the downstream services first. Then run
```
Comment (johno1985) [inline README.md:21]: typo

**Review by johno1985 — COMMENTED** [review]
comments added

**Comment by johno1985** [comment]
looks like there are some merge conflicts, rebase required

**Review by brdgeth — APPROVED** [review]

**Pull request merged**